File: sample/src/main/java/com/infullmobile/android/infullmvp/sample/fragment/SampleFragmentView.java
package com.infullmobile.android.infullmvp.sample.fragment;

import android.support.annotation.LayoutRes;
import android.widget.TextView;

import com.infullmobile.android.infullmvp.PresentedFragmentView;
import com.infullmobile.infullmvp.sample.R;

public class SampleFragmentView extends PresentedFragmentView<SampleFragmentPresenter> {

    private TextView fragmentTextViewContainer;

    @LayoutRes
    @Override
    protected int getLayoutResId() {
        return R.layout.fragment_sample;
    }

    @Override
    protected void bindViews() {
        fragmentTextViewContainer = findView(R.id.fragmentTextContainer);
    }

    @Override
    protected void onViewsBound() {
    }

    public void setText(String message) {
        fragmentTextViewContainer.setText(message);
    }
}

File: sample-kotlin/src/main/java/com/infullmobile/android/infullmvp/sample/fragment/SampleFragmentView.kt
package com.infullmobile.android.infullmvp.sample.fragment

import android.support.annotation.LayoutRes
import android.widget.Button
import android.widget.TextView
import com.infullmobile.android.infullmvp.PresentedFragmentView
import com.infullmobile.android.infullmvp.sample.R

class SampleFragmentView : PresentedFragmentView<SampleFragmentPresenter>() {

    @LayoutRes
    override val layoutResId = R.layout.fragment_sample
    private val fragmentTextViewContainer: TextView by bindView(R.id.fragmentTextContainer)
    private val relaunchButton: Button by bindView(R.id.relaunchButton)

    override fun onViewsBound() {
        relaunchButton.setOnClickListener { presenter.relaunchButtonClicked() }
    }

    fun setText(message: String) {
        fragmentTextViewContainer.text = message
    }
}
